server.py

```diff
@@ -169,14 +173,46 @@ def can_move_file():
     space_left = int(storage_limit) - current_storage
     response_message = space_left
     if file_size < space_left:
-        return str(response_message), requests.code.ok
+        return str(response_message), requests.codes.ok
     else:
-        return str(response_message), requests.code.request_entity_too_large
+        return str(response_message), requests.codes.request_entity_too_large
 
 # Returns the capacity of the server
 @app.route('/capacity', methods=['GET'])
 def capacity():
-    return str(app.config['storage_limit']), requests.code.ok
+    return str(app.config['storage_limit']), requests.codes.ok
+
+# Updates and retrieves metadata for a file
+@app.route('/metadata', methods=['GET'])
+def metadata():
+    response = None
+
+    filename = request.args.get('uuid')
+
+    file_info = metadata.is_file_exist_locally(filename, app.config['HOST'])
+    
+    if file_info is not None:
+        response = { 'uuid': file_info[0], 'server': file_info[1], 'file_size': file_info[2] }
+    else:
+        server_with_file = metadata.lookup_file(filename, app.config['HOST'])
+        
+        if server_with_file is None:
+            other_servers = metadata.get_all_server(app.config['HOST'])
+            if (len(other_servers) > 0):
+                for server in other_servers:
+                    url = 'http://%s/file_exists?%s' % (server, urllib.urlencode({ 'uuid': filename }))
+                    lookup_request = requests.get(url)
+                    if (lookup_request.status_code == requests.codes.ok):
+                        server_with_file = server
+                        break
+
+        if server_with_file is not None:
+            response = update_metadata_from_another_server(server_with_file, filename)
+
+    if response is None:
+        return 'File not found', requests.codes.not_found
+
+    return json.dumps(response), requests.codes.ok
```

Comment:
Nope actually, JSON is also my fav data format.
On Mar 27, 2015 10:05 AM, "Chris Baik" <notifications@github.com> wrote:

> In server.py
> <https://github.com/paivaspol/EECS591/pull/57#discussion_r27296713>:
>
> > +            other_servers = metadata.get_all_server(app.config['HOST'])
> > +            if (len(other_servers) > 0):
> > +                for server in other_servers:
> > +                    url = 'http://%s/file_exists?%s' % (server, urllib.urlencode({ 'uuid': filename }))
> > +                    lookup_request = requests.get(url)
> > +                    if (lookup_request.status_code == requests.codes.ok):
> > +                        server_with_file = server
> > +                        break
> > +
> > +        if server_with_file is not None:
> > +            response = update_metadata_from_another_server(server_with_file, filename)
> > +
> > +    if response is None:
> > +        return 'File not found', requests.codes.not_found
> > +
> > +    return json.dumps(response), requests.codes.ok
>
> yeah, my preferred data format tool. any suggested alternatives?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paivaspol/EECS591/pull/57/files#r27296713>.
>